sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/models/AmqpAnnotatedMessage.java

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.amqp.models;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Amqp representation of the message.
+ */
+public final class AmqpAnnotatedMessage {
+    private final AmqpMessageBody amqpMessageBody;
+    private Map<String, Object> applicationProperties;
+    private Map<String, Object> deliveryAnnotations;
+    private Map<String, Object> messageAnnotations;
+    private Map<String, Object> footer;
```

Comment:
It is because unlike `AmqpMessageHeader` , The footer could be represented by `Map<String, Object>`. So we do not need another type for this. We are exposing internal representation as it is.